packages/core/src/crypto/JwsService.ts

```diff
@@ -92,28 +131,33 @@ export class JwsService {
     return { isValid: true, signerVerkeys }
   }
 
-  /**
-   * @todo This currently only work with a single alg, key type and curve
-   *    This needs to be extended with other formats in the future
-   */
-  private buildProtected(verkey: string) {
+  private buildProtected(verkey: Key, options: ProtectedHeaderOptions) {
+    if (!options.jwk && !options.kid) {
+      throw new AriesFrameworkError('Both JWK and kid are undefined. Please provide one or the other.')
+    }
+    if (options.jwk && options.kid) {
+      throw new AriesFrameworkError('Both JWK and kid are provided. Please only provide one of the two.')
+    }
+
     return {
-      alg: 'EdDSA',
-      jwk: {
-        kty: 'OKP',
-        crv: 'Ed25519',
-        x: TypedArrayEncoder.toBase64URL(TypedArrayEncoder.fromBase58(verkey)),
-      },
+      alg: options.alg,
```

Comment:
We don't check whether the `alg` is valid for the provided `KeyType`. Is that up to the user?